fdbserver/DataDistributionTracker.actor.cpp

```diff
@@ -546,30 +546,60 @@ ACTOR Future<Void> shardSplitter(DataDistributionTracker* self,
 		int skipRange = deterministicRandom()->randomInt(0, numShards);
 		// The queue can't deal with RelocateShard requests which split an existing shard into three pieces, so
 		// we have to send the unskipped ranges in this order (nibbling in from the edges of the old range)
-		for (int i = 0; i < skipRange; i++)
-			restartShardTrackers(self, KeyRangeRef(splitKeys[i], splitKeys[i + 1]));
-		restartShardTrackers(self, KeyRangeRef(splitKeys[skipRange], splitKeys[skipRange + 1]));
-		for (int i = numShards - 1; i > skipRange; i--)
-			restartShardTrackers(self, KeyRangeRef(splitKeys[i], splitKeys[i + 1]));
-
-		for (int i = 0; i < skipRange; i++) {
-			KeyRangeRef r(splitKeys[i], splitKeys[i + 1]);
-			self->shardsAffectedByTeamFailure->defineShard(r);
-			self->output.send(RelocateShard(r, priority));
-		}
-		for (int i = numShards - 1; i > skipRange; i--) {
-			KeyRangeRef r(splitKeys[i], splitKeys[i + 1]);
-			self->shardsAffectedByTeamFailure->defineShard(r);
-			self->output.send(RelocateShard(r, priority));
-		}
-		if (doManualSplit) {
-			// If manual shard split, we move all shards out of the team
-			KeyRangeRef r(splitKeys[skipRange], splitKeys[skipRange + 1]);
-			self->shardsAffectedByTeamFailure->defineShard(r);
-			self->output.send(RelocateShard(r, priority));
-		}
+		if (SERVER_KNOBS->EMERGENCY_DISABLE_DATA_MOVE) {
+			for (int i = 0; i < skipRange; i++) {
+				KeyRangeRef r(splitKeys[i], splitKeys[i + 1]);
+				self->output.send(RelocateShard(r, priority));
```

Comment:
Why do we use RelocateShard in this case?

Reply:
same reason.